### PR TITLE
CLN: Assorted cleanups

### PR DESCRIPTION
**Pull request by jbrockmendel**

### CODE REVIEW

File: pandas/_libs/internals.pyx [hunk 1]
```diff
@@ -284,7 +284,7 @@ cdef slice_get_indices_ex(slice slc, Py_ssize_t objlen=PY_SSIZE_T_MAX):
     return start, stop, step, length
 
 
-def slice_getitem(slice slc not None, ind):
+cdef slice_getitem(slice slc, ind):
```
Comment (WillAyd) [inline pandas/_libs/internals.pyx]: Do we have a `not None` check elsewhere?

**Review by WillAyd — COMMENTED** [review]
looks good minor question

File: pandas/_libs/internals.pyx (hunk 1 above)
Comment (jbrockmendel) [reply pandas/_libs/internals.pyx]: `not None` is not allowed in a cdef function.  we only call this from one place in this module, and it is immediately after checking that we do in fact have a slice

**Comment by jreback** [comment]
thanks

**Pull request merged**